components/applications-service/integration_test/disconnected_services_job_test.go

```diff
@@ -109,8 +112,96 @@ func TestPeriodicDisconnectedServices(t *testing.T) {
+
+		require.NoError(t, err)
+		assert.Equal(t, "delete_disconnected_services", sched.WorkflowName)
+		assert.True(t, sched.Enabled)
+	})
+
+	t.Run("disable and enable delete_disconnected_services job", func(t *testing.T) {
+		req := &applications.PeriodicJobConfig{Threshold: "7d", Running: false}
+		_, err := suite.ApplicationsServer.UpdateDeleteDisconnectedServicesConfig(ctx, req)
+		require.NoError(t, err)
+
+		conf, err := suite.ApplicationsServer.GetDeleteDisconnectedServicesConfig(ctx, &applications.GetDeleteDisconnectedServicesConfigReq{})
+		require.NoError(t, err)
+		assert.False(t, conf.Running)
+
+		req.Running = true
+		_, err = suite.ApplicationsServer.UpdateDeleteDisconnectedServicesConfig(ctx, req)
+		require.NoError(t, err)
+
+		conf, err = suite.ApplicationsServer.GetDeleteDisconnectedServicesConfig(ctx, &applications.GetDeleteDisconnectedServicesConfigReq{})
+		require.NoError(t, err)
+		assert.True(t, conf.Running)
+	})
+
+	t.Run("update delete_disconnected_services job params", func(t *testing.T) {
+		defer suite.JobScheduler.ResetParams()
+
+		req := &applications.PeriodicJobConfig{Threshold: "23s", Running: true}
+		_, err := suite.ApplicationsServer.UpdateDeleteDisconnectedServicesConfig(ctx, req)
+		require.NoError(t, err)
+
+		conf, err := suite.ApplicationsServer.GetDeleteDisconnectedServicesConfig(ctx, &applications.GetDeleteDisconnectedServicesConfigReq{})
+		require.NoError(t, err)
+		assert.Equal(t, "23s", conf.Threshold)
+
+		// Update the params again to ensure we didn't accidentally pass the test due to leftover state somewhere:
+		req = &applications.PeriodicJobConfig{Threshold: "42s", Running: true}
+		_, err = suite.ApplicationsServer.UpdateDeleteDisconnectedServicesConfig(ctx, req)
+		require.NoError(t, err)
+
+		conf, err = suite.ApplicationsServer.GetDeleteDisconnectedServicesConfig(ctx, &applications.GetDeleteDisconnectedServicesConfigReq{})
+		require.NoError(t, err)
+		assert.Equal(t, "42s", conf.Threshold)
+	})
+
+	t.Run("running the job runner makes the delete_disconnected_services job run", func(t *testing.T) {
+		err := suite.JobScheduler.RunAllJobsConstantly(ctx)
+		require.NoError(t, err)
+
+		// * have a way to track number of job runs (prometheus (?))
+		defer suite.DeleteDataFromStorage()
+
+		event := NewHabitatEvent(
+			withSupervisorId("abcd"),
+			withServiceGroup("postgres.default"),
+			withPackageIdent("core/postgres/0.1.0/20190101121212"),
+			withStrategyAtOnce("testchannel"),
+			withFqdn("mytest.example.com"),
+			withSite("testsite"),
+		)
+
+		// Patch event timestamp to mock an old service message which we'll delete for being disconnected
+		event.EventMetadata.OccurredAt, err = ptypes.TimestampProto(time.Now().Add(-time.Hour * 24 * 8))
+		require.NoError(t, err)
+		suite.IngestService(event)
+
+		logrus.Info("Starting check for job runner")
+		runsThusFar := runners.DeleteDisconnectedServicesExecutor.TotalRuns()
+		detectedJobRun := false
+		for i := 0; i <= 100; i++ {
```

Comment:
Interesting way to solve the eventual consistency issue. I like it, though perhaps we should be looking for `TotalRuns() == runsThusFar +1`